common/Log.hpp

```diff
@@ -282,22 +294,35 @@ static constexpr std::size_t skipPathPrefix(const char (&s)[N], std::size_t n =
         LOG.flush();                                                                               \
     } while (false)
 
-#define LOG_BODY_(LOG, PRIO, LVL, X, PREFIX, END)                                                  \
+#define LOG_BODY__(LVL, X, TYPE, PREFIX, SUFFIX)                                                   \
+    do                                                                                             \
+    {                                                                                              \
+        auto& log_ = Log::logger();                                                                \
+        if (LOG_CONDITIONAL(log_, TYPE))                                                           \
+        {                                                                                          \
+            LOG_BODY_(log_, LVL, X, PREFIX, SUFFIX);                                               \
+        }                                                                                          \
+    } while (false)
+
+
+#define LOG_BODY_(LOG, LVL, X, PREFIX, END)                        \
     char b_[1024];                                                                                 \
-    std::ostringstream oss_(Log::prefix<sizeof(b_) - 1>(b_, LVL), std::ostringstream::ate);        \
+    std::ostringstream oss_(Log::prefix<sizeof(b_) - 1>(b_, #LVL), std::ostringstream::ate);       \
     PREFIX(oss_);                                                                                  \
     oss_ << std::boolalpha << X;                                                                   \
     END(oss_);                                                                                     \
-    LOG_LOG(LOG, PRIO, LVL, oss_.str())
+    LOG_LOG(LOG, LOG.name(), LVL, oss_.str())
 
 #define LOG_ANY(X)                                                                                 \
-    char b_[1024];                                                                                 \
-    std::ostringstream oss_(Log::prefix<sizeof(b_) - 1>(b_, "INF"), std::ostringstream::ate);      \
-    logPrefix(oss_);                                                                               \
-    oss_ << std::boolalpha << X;                                                                   \
-    LOG_END(oss_);                                                                                 \
-    Poco::AutoPtr<Poco::Channel> channel = Log::logger().getChannel();                             \
```

Comment:
Perhaps the better fix is to check that `channel` here is not null?

Reply:
I did it in my initial patches, if you can check the history